python/cudf/cudf/core/dataframe.py

```diff
@@ -7297,3 +7324,22 @@ def _get_host_unique(array):
         return [array]
     else:
         return set(array)
+
+
+def _reindex_by_columns(
```

Comment:
Rather than writing utilities that operate on column accessors/dictionaries, which are very much internal details of frames, I'd suggest that they operate on frames instead. In the future, if we change the internals of frames, the implementation of the utility will change, but callers won't need to worry about it.

That's why I'd recommend a `_reindex_frame()` utility over a `_reindex_internal_data()` utility.